### PR TITLE
fix: [CDS-84788]: adds validation for InputSet Identifier on create and update

### PR DESCRIPTION
**Pull request by sandeshsalunkhegh**
fix: [CDS-84788]: adds validation for InputSet Identifier on create and update

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-84788]: https://harness.atlassian.net/browse/CDS-84788?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55642)

### CODE REVIEW

File: pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/InputSetValidationHelper.java [hunk 1]
```diff
@@ -67,6 +70,11 @@ public void validateInputSet(
       default:
         throw new IllegalStateException("version not supported");
     }
+    String inputSetIdentifier = inputSetEntity.getIdentifier();
```
Comment (shivagowda) [inline pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/InputSetValidationHelper.java:73]: can you add unit test for this?

Comment (sandeshsalunkhegh) [reply pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/InputSetValidationHelper.java:73]: Yes @shivagowda , I will add the unit test as well once the changes are finalised.

**Comment by sandeshsalunkhegh** [comment]
With the Feature Flag Enabled
image
With the Feature Flag Disabled
image
image

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/InputSetValidationHelper.java` line 96 at r5](https://reviewable.io/reviews/harness/harness-core/55642#-NjlYci0A_dX1tndAAyB:-NjlYci0A_dX1tndAAyC:blk0i2p) ([raw file](https://github.com/harness/harness-core/blob/92754563accf70cb8bac6efda776c4401b7e466c/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/InputSetValidationHelper.java#L96)):*
> ```Java
>       }
>     } else {
>       log.warn(format("InputSet Identifier cannot contain special characters or spaces: [%s]", inputSetIdentifier));
> ```

We will analyse this and if we don't find this log in the next 2 weeks, we will enable the FF

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 4 files at r5.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55642)* status: 4 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by adivishy1 — APPROVED** [review]
> Reviewed 3 of 4 files at r5, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55642)* status: 4 file(s) to review. 1. unresolved discussions.

**Pull request merged**